packager/VisibilityChecker.cc

```diff
@@ -421,6 +421,57 @@ const FileType fileTypeFromCtx(const core::Context ctx) {
     }
 }
 
+void untrackPackageReferences(core::packages::PackageDB &nonConstPackageDB, absl::Span<ast::ParsedFile> filesSpan) {
+    const core::packages::PackageDB &packageDB = nonConstPackageDB;
+    for (auto &parsedFile : filesSpan) {
+        // TODO(neil): we can skip this loop on the slow path since the package references sets will be
+        // empty
+        auto file = parsedFile.file;
+        auto pkgName = packageDB.getPackageNameForFile(file);
+        if (!pkgName.exists()) {
+            continue;
+        }
+        auto nonConstPackageInfo = nonConstPackageDB.getPackageInfoNonConst(pkgName);
+        nonConstPackageInfo->untrackPackageReferencesFor(file);
+    }
+}
+
+void untrackSymbolReferences(core::GlobalState &nonConstGs, absl::Span<ast::ParsedFile> filesSpan) {
+    const core::GlobalState &gs = nonConstGs;
+    std::shared_ptr<core::FileTable> fileTable = nonConstGs.getFileTable();
+    for (auto &parsedFile : filesSpan) {
+        // TODO(neil): we can skip this loop on the slow path since the package references sets will be
+        // empty
+        auto file = parsedFile.file;
+        if (fileTable->getSymbolsReferencedByFile(file.id()).size() == 0) {
+            continue;
+        }
+        for (auto &symbol : fileTable->getSymbolsReferencedByFile(file.id())) {
+            bool deletedFile = false;
+            if (symbol.isClassOrModule()) {
+                auto &files = symbol.asClassOrModuleRef().data(nonConstGs)->referencingFiles;
+                auto it = files.find(file);
+                if (it != files.end()) {
+                    files.erase(it);
+                    deletedFile = true;
+                }
+            } else if (symbol.isFieldOrStaticField()) {
+                auto &files = symbol.asFieldRef().data(nonConstGs)->referencingFiles;
+                auto it = files.find(file);
+                if (it != files.end()) {
+                    files.erase(it);
+                    deletedFile = true;
+                }
+            } else {
+                ENFORCE(false);
+                continue;
+            }
+            ENFORCE(deletedFile, "{} is in symbolsReferencedByFile for {} but not file is not in referencingFiles",
+                    symbol.show(gs), file.data(gs).path());
+        }
+        fileTable->getSymbolsReferencedByFile(file.id()).clear();
```

Comment:
`getSymbolsReferencedByFile` does not return a reference, so this `.clear()` is clearing a copy of the Set. That's probably not what we intended?